pvlib/iotools/srml.py

```diff
@@ -0,0 +1,148 @@
+    on a given line should now be understood to occur during the interval
+    extending from the time of the line in which they are listed to
+    the ending time on the next line, rather than the previous line.
+    """
+    tsv_data = pd.read_csv(filename, delimiter='\t')
+    year = tsv_data.columns[1]
+    data = format_index(tsv_data, year)
+    # Rename and drop datetime columns
+    data = data[data.columns[2:]].rename(columns=map_columns)
+
+    # Quality flags are all labeled 0, but occur immediately after their
+    # associated var so we create a dict mapping them to var_flag for renaming
+    columns = data.columns
+    flag_label_map = {flag: columns[columns.get_loc(flag) - 1] + '_flag'
+                      for flag in columns[1::2]}
+    data = data.rename(columns=flag_label_map)
+    # For data flagged bad or missing, replace the value with np.NaN
+    for col in columns[::2]:
+        data[col] = data[col].where(~(data[col + '_flag'] == 99), np.NaN)
+    return data
+
+
+def map_columns(col):
+    """Map column labels to pvlib names.
+
+    Parameters
+    ----------
+    col: str
+        Column label to be mapped.
+
+    Returns
+    -------
+    str
+        The pvlib label if it was found in the mapping,
+        else the original label.
+
+    Notes
+    -----
+    var_map is a dictionary mapping SRML data element numbers
+    to their pvlib names. For most variables, only the first
+    three digits are used, the fourth indicating the instrument.
+    Spectral data (7xxx) uses all four digits to indicate the
+    variable.
+    """
+    var_map = {
+        '100': 'ghi',
+        '201': 'dni',
+        '300': 'dhi',
+        '920': 'wind_dir',
+        '921': 'wind_speed',
+        '930': 'temp_air',
+        '931': 'temp_dew',
+        '933': 'relative_humidity',
+        '937': 'temp_cell',
+    }
+    if col.startswith('7'):
+        # spectral data
+        try:
+            return var_map[col]
+        except KeyError:
+            return col
+    try:
+        return var_map[col[:3]] + '_' + col[3:]
+    except KeyError:
+        return col
+
+
+def format_index(df, year):
+    """ Create a datetime index from day of year, and time columns.
+
+    Parameters
+    ----------
+    df: pd.Dataframe
+        The srml data to reindex.
+    year: int
+        The year of the file
+
+    Returns
+    -------
+    df: pd.Dataframe
+        The Dataframe with a datetime index applied.
+    """
+    df_time = df[df.columns[1]] - 1
+    df_doy = df[df.columns[0]]
+    fifty_nines = df_time % 100 == 99
+    times = df_time.where(~fifty_nines, df_time - 40)
+    times = times.apply(lambda x: '{:04.0f}'.format(x))
+    doy = df_doy.apply(lambda x: '{:03.0f}'.format(x))
+    dts = pd.to_datetime(str(year) + '-' + doy + '-' + times,
+                         format='%Y-%j-%H%M')
+    df.index = dts
+    df = df.tz_localize('Etc/GMT+8')
+    return df
+
+
+def request_srml_data(station, year, month, filetype='PO'):
```

Comment:
maybe `read_srml_from_solardat` and import it in `__init__.py`?

Reply:
Should this module include these kind of util functions? I wasn't sure about including this here, or a function to stitch together due to their function signatures being pretty far off from read_<data_type> functions.

Reply:
I think the module should include any generally useful functions for IO related to the UOSRML. That would include downloading directly from their website. 

I suppose this function might more accurately be called `read_srml_month_from_solardat`

I'm less sure about a function to download and stitch together multiple months because you said that the columns sometimes change. If we made it, that's the function that would be called `read_srml_from_solardat`